crates/server/src/authority/catalog.rs

```diff
@@ -421,22 +438,30 @@ async fn lookup<'a, R: ResponseHandler + Unpin>(
     )
     .await;
 
-    let response = MessageResponseBuilder::new(Some(request.raw_query())).build(
-        response_header,
-        sections.answers.iter(),
-        sections.ns.iter(),
-        sections.soa.iter(),
-        sections.additionals.iter(),
-    );
+    match response {
```

Comment:
Style nit: to reduce rightward drift, it would be nice to have early returns for the `Err` and `Skip` cases, and yield the `lookup` out of the `match` for further handling.

For a bunch of these `match`es, maybe `use LookupResult::*` just before the `match` to reduce the noise a bit?

Reply:
@djc I've simplified some of the matches and brought LookupControlFlow::* into scope to reduce the noise a bit.  Please take a look when you have time and let me know if you have any other feedback.

Reply:
@djc, any remaining concern here?